README.md

```diff
@@ -1,96 +1,354 @@
+1. it must return a valid fragment, if and only if the document data hold a name property with the value `Certificate of Completion`
+
+**Document version must be equal to `https://schema.openattestation.com/2.0/schema.json`**
+
+This is where `skip` and `test` methods come into play. We will use the `test` method to return when the verification method run, and the `skip` method to explain why it didn't run:
+
+```ts
+// index.ts
+import { verificationBuilder, openAttestationVerifiers, isValid } from "@govtechsg/oa-verify";
+import { getData } from "@govtechsg/open-attestation";
+import * as document from "./document.json";
+
+const customVerifier = {
+  skip: async () => {
+    return {
+      status: "SKIPPED",
+      type: "DOCUMENT_INTEGRITY",
+      name: "CustomVerifier",
+      reason: {
+        code: 0,
+        codeString: "SKIPPED",
+        message: `Document doesn't have version equal to 'https://schema.openattestation.com/2.0/schema.json'`,
+      },
+    };
+  },
+  test: () => document.version === "https://schema.openattestation.com/2.0/schema.json",
+};
+```
+
+> we use `DOCUMENT_INTEGRITY` type because we check for the content of the document.
+
+**Document holds correct `name` property**
+
+Once we have decided `when` the verification method run, it's time to write the logic of the verifier in the `verify` method. We will use [getData](https://www.openattestation.com/docs/developer-section/libraries/open-attestation#retrieving-document-data) utility to access the data of the document and return the appropriate fragment depending on the content:
+
+```ts
+// index.ts
+import { verificationBuilder, openAttestationVerifiers, isValid } from "@govtechsg/oa-verify";
+import { getData } from "@govtechsg/open-attestation";
+import * as document from "./document.json";
+
+const customVerifier = {
+  skip: async () => {
+    /* content has been defined in the section above */
+  },
+  test: () => /* content has been defined in the section above */,
+  verify: async (document: any) => {
+    const documentData = getData(document);
+    if (documentData.name !== "Certificate of Completion") {
+      return {
+        type: "DOCUMENT_INTEGRITY",
+        name: "CustomVerifier",
+        data: documentData.name,
+        reason: {
+          code: 1,
+          codeString: "INVALID_NAME",
+          message: `Document name is ${documentData.name}`,
+        },
+        status: "INVALID",
+      };
+    }
+    return {
+      type: "DOCUMENT_INTEGRITY",
+      name: "CustomVerifier",
+      data: documentData.name,
+      status: "VALID",
+    };
+  },
+};
+```
+
+#### Building a custom verify method
+
+The `verify` function is built to run a list of verification method. Each verifier will produce a fragment that will help to determine if the document is valid. OpenAttestation comes with its own set of verification methods available in `openAttestationVerifiers`.
```

Comment:
The two first paragraph looks redundant with what we explained above. Maybe can reference to above instead